lib/Parse/ParseDecl.cpp

```diff
@@ -2536,49 +2536,66 @@ ParsedSyntaxResult<ParsedAttributeSyntax> Parser::parseTypeAttributeSyntax() {
   }
 
   // SWIFT_ENABLE_TENSORFLOW
-  case TAK_differentiable: {
-    bool linear = false;
-    // Check if there is a 'linear' argument.
-    if (Tok.is(tok::l_paren) && peekToken().is(tok::identifier)) {
+  case TAK_differentiable:
+    status |= [&]() -> ParserStatus {
+      bool linear = false;
+      // Check if there is a 'linear' argument.
+      // If next tokens are not `'(' identifier`, break early.
+      if (!Tok.is(tok::l_paren) || !peekToken().is(tok::identifier))
+        return makeParserSuccess();
+
       Parser::BacktrackingScope backtrack(*this);
-      consumeToken(tok::l_paren);
+      SourceLoc LParenLoc = Tok.getLoc();
```

Comment:
Minor suggestion for camelcase consistency, same for `RParenLoc` below.
```suggestion
      SourceLoc lParenLoc = Tok.getLoc();
```

Reply:
done